pkg/assetfetcher/errors.go

```diff
@@ -51,12 +51,14 @@ func (e *NexposeHTTPRequestError) Error() string {
 
 // ErrorFetchingAssets represents an error we get when trying to fetch assets
 type ErrorFetchingAssets struct {
-	Inner error
+	Inner  error
+	Page   int
+	SiteID string
 }
 
 // Error returns an ErrorFetchingAssets
 func (e *ErrorFetchingAssets) Error() string {
-	return fmt.Sprintf("error fetching assets from Nexpose %v", e.Inner)
+	return fmt.Sprintf("error fetching assets site: %v, on page %d, from Nexpose. %v", e.SiteID, e.Page, e.Inner)
```

Comment:
Task: Would it make more sense for these fields to be log fields instead of in the error message? I'm thinking about the "message" groupings in splunk and being able to see all similar errors grouped together rather than having unique error messages for each site/page.

Reply:
Yes, this! I don't think this team has explicitly standardized on that technique of using static, searchable and group-able, message fields with dynamic specific values for each occurrence, but it is a good one. I have found it incredibly helpful in the past when troubleshooting running services.

So yes, please leave the site and page out of the message, and ensure that the emitted log line contains the structured field data.

Reply:
can do